### PR TITLE
feat(pre-commit-hook): adding commitlint support

### PR DESCRIPTION
**Pull request by fruch**
See [conventinal commits](https://www.conventionalcommits.org/en/v1.0.0/) for more inforamtion
on rules and how we can configure it

trello: https://trello.com/c/iqaQIfQQ

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- ~~[ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -6,28 +6,28 @@ def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
...
+    }
+    sh """
+    #!/bin/bash
+    set -xe
+    env
+    export SCT_CLUSTER_BACKEND="${backend}"
+    export SCT_COLLECT_LOGS=false
     export SCT_CONFIG_FILES="test-cases/PR-provision-test.yaml"
     export BUILD_USER_ID="\${CHANGE_AUTHOR}"
-	$distro_cmd
```
Comment (fruch) [inline Jenkinsfile]: fixing mixed tabs/spaces in this file

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
+    rules: {
+        'body-leading-blank': [1, 'always'],
+        'footer-leading-blank': [1, 'always'],
+        'header-max-length': [2, 'always', 72],
```
Comment (bentsi) [inline commitlint.config.js]: `72` -> `85`

Comment (fruch) [reply commitlint.config.js]: fixed

File: commitlint.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
```
Comment (bentsi) [inline commitlint.config.js]: I am working on the rules definitions, will send email to Roy for approval

**Comment by bentsi** [comment]
@fruch please replace `commitlint.config.js` with the following config:
```
module.exports = {
    rules: {
        // Subject
        'subject-empty': [2, 'never'],
        'subject-full-stop': [2, 'never', '.'],
        'subject-max-length': [2, 'always', 85],
        'subject-min-length': [2, 'always', 10],
        // Type
        'type-enum': [2,'always',['ci','docs','feature','fix','improvement','perf','refactor','revert','style','test', 'unit-test']],
        'type-empty': [2, 'never'],
        // Scope
        'scope-empty': [1, 'never'],
        'scope-case': [2, 'always', ['lower-case']],
        'scope-min-length': [2, 'always', 5],
        // Body
        'body-min-length': [2, 'always', 30],
        'body-max-line-length': [2, 'always', 100],
        'body-leading-blank': [2, 'always'],
    }
};

```

**Pull request merged**